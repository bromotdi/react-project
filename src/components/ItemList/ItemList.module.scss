@import '../../scss/mixins';
@import '../../scss/variables';

.itemList {
  margin-bottom: 3rem;

  &__wrapper {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    gap: 2rem;
    padding: 0 1rem;
  }

  &__title {
    @include font(3.6rem, 700, $text-color-main);
    margin: 0 0 2rem 4rem;
  }
}
