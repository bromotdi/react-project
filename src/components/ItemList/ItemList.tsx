import React from 'react';
import { PersonCard } from './PersonCard';
import { UserCard } from './UserCard';
import { Person, User } from '../../types';

import style from './ItemList.module.scss';

interface ItemListProps {
  persons?: Person[];
  users?: User[];
}

export class ItemList extends React.Component<ItemListProps> {
  constructor(props: ItemListProps) {
    super(props);
  }

  render() {
    return (
      <section className={style.itemList}>
        <h1 className={style.itemList__title}>{this.props.persons ? 'Persons' : 'Users'}</h1>

        <div className={style.itemList__wrapper}>
          {this.props.persons?.map((item) => (
            <PersonCard key={item.id} person={item} />
          ))}
          {this.props.users?.map((item) => (
            <UserCard key={item.id} user={item} />
          ))}
        </div>
      </section>
    );
  }
}
