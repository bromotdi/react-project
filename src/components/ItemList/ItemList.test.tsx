import { render, screen } from '@testing-library/react';
import { ItemList } from './ItemList';

const testCards = [
  {
    id: 1,
    name: 'Eren Yeager',
    status: 'Deceased',
    species: 'Human, Intelligent Titan',
    gender: 'Male',
    location: 'Wall Rose',
    image:
      'https://static.wikia.nocookie.net/shingekinokyojin/images/a/a1/Eren_Jaeger_%28Anime%29_character_image.png',
  },
  {
    id: 2,
    name: 'Mikasa Ackermann',
    status: 'Alive',
    species: 'Human',
    gender: 'Female',
    location: 'Shiganshina District',
    image:
      'https://static.wikia.nocookie.net/shingekinokyojin/images/4/4f/Mikasa_Ackermann_%28Anime%29_character_image.png',
  },
  {
    id: 3,
    name: 'Armin Arlelt',
    status: 'Alive',
    species: 'Human, Intelligent Titan',
    gender: 'Male',
    location: 'Shiganshina District',
    image:
      'https://static.wikia.nocookie.net/shingekinokyojin/images/9/93/Armin_Arlelt_%28Anime%29_character_image.png',
  },
];
describe('ItemList', () => {
  it('render elements in ItemList', () => {
    const { container } = render(<ItemList persons={testCards} />);

    expect(screen.getByText(/Persons/i)).toBeInTheDocument();
    expect(container.getElementsByClassName('person').length).toBe(testCards.length);
  });
});
