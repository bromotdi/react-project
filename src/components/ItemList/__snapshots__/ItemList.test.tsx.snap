// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList ItemList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="itemList"
      >
        <h1
          class="itemList__title"
        >
          Persons
        </h1>
        <div
          class="itemList__wrapper"
        >
          <div
            class="person"
          >
            <img
              class="person__img"
              src="https://static.wikia.nocookie.net/shingekinokyojin/images/a/a1/Eren_Jaeger_%28Anime%29_character_image.png"
            />
            <div
              class="person__info"
            >
              <h2
                class="person__name"
              >
                Eren Yeager
              </h2>
              <p
                class="person__species"
              >
                <span
                  class="person__type"
                >
                  species: 
                </span>
                <span
                  class="person__text"
                >
                  Human, Intelligent Titan
                </span>
              </p>
              <p
                class="person__gender"
              >
                <span
                  class="person__type"
                >
                  gender: 
                </span>
                <span
                  class="person__text"
                >
                  Male
                </span>
              </p>
              <p
                class="person__status"
              >
                <span
                  class="person__type"
                >
                  status: 
                </span>
                <span
                  class="person__text"
                >
                  Deceased
                </span>
              </p>
              <p
                class="person__location"
              >
                <span
                  class="person__type"
                >
                  location: 
                </span>
                <span
                  class="person__text"
                >
                  Wall Rose
                </span>
              </p>
            </div>
          </div>
          <div
            class="person"
          >
            <img
              class="person__img"
              src="https://static.wikia.nocookie.net/shingekinokyojin/images/4/4f/Mikasa_Ackermann_%28Anime%29_character_image.png"
            />
            <div
              class="person__info"
            >
              <h2
                class="person__name"
              >
                Mikasa Ackermann
              </h2>
              <p
                class="person__species"
              >
                <span
                  class="person__type"
                >
                  species: 
                </span>
                <span
                  class="person__text"
                >
                  Human
                </span>
              </p>
              <p
                class="person__gender"
              >
                <span
                  class="person__type"
                >
                  gender: 
                </span>
                <span
                  class="person__text"
                >
                  Female
                </span>
              </p>
              <p
                class="person__status"
              >
                <span
                  class="person__type"
                >
                  status: 
                </span>
                <span
                  class="person__text"
                >
                  Alive
                </span>
              </p>
              <p
                class="person__location"
              >
                <span
                  class="person__type"
                >
                  location: 
                </span>
                <span
                  class="person__text"
                >
                  Shiganshina District
                </span>
              </p>
            </div>
          </div>
          <div
            class="person"
          >
            <img
              class="person__img"
              src="https://static.wikia.nocookie.net/shingekinokyojin/images/9/93/Armin_Arlelt_%28Anime%29_character_image.png"
            />
            <div
              class="person__info"
            >
              <h2
                class="person__name"
              >
                Armin Arlelt
              </h2>
              <p
                class="person__species"
              >
                <span
                  class="person__type"
                >
                  species: 
                </span>
                <span
                  class="person__text"
                >
                  Human, Intelligent Titan
                </span>
              </p>
              <p
                class="person__gender"
              >
                <span
                  class="person__type"
                >
                  gender: 
                </span>
                <span
                  class="person__text"
                >
                  Male
                </span>
              </p>
              <p
                class="person__status"
              >
                <span
                  class="person__type"
                >
                  status: 
                </span>
                <span
                  class="person__text"
                >
                  Alive
                </span>
              </p>
              <p
                class="person__location"
              >
                <span
                  class="person__type"
                >
                  location: 
                </span>
                <span
                  class="person__text"
                >
                  Shiganshina District
                </span>
              </p>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="itemList"
    >
      <h1
        class="itemList__title"
      >
        Persons
      </h1>
      <div
        class="itemList__wrapper"
      >
        <div
          class="person"
        >
          <img
            class="person__img"
            src="https://static.wikia.nocookie.net/shingekinokyojin/images/a/a1/Eren_Jaeger_%28Anime%29_character_image.png"
          />
          <div
            class="person__info"
          >
            <h2
              class="person__name"
            >
              Eren Yeager
            </h2>
            <p
              class="person__species"
            >
              <span
                class="person__type"
              >
                species: 
              </span>
              <span
                class="person__text"
              >
                Human, Intelligent Titan
              </span>
            </p>
            <p
              class="person__gender"
            >
              <span
                class="person__type"
              >
                gender: 
              </span>
              <span
                class="person__text"
              >
                Male
              </span>
            </p>
            <p
              class="person__status"
            >
              <span
                class="person__type"
              >
                status: 
              </span>
              <span
                class="person__text"
              >
                Deceased
              </span>
            </p>
            <p
              class="person__location"
            >
              <span
                class="person__type"
              >
                location: 
              </span>
              <span
                class="person__text"
              >
                Wall Rose
              </span>
            </p>
          </div>
        </div>
        <div
          class="person"
        >
          <img
            class="person__img"
            src="https://static.wikia.nocookie.net/shingekinokyojin/images/4/4f/Mikasa_Ackermann_%28Anime%29_character_image.png"
          />
          <div
            class="person__info"
          >
            <h2
              class="person__name"
            >
              Mikasa Ackermann
            </h2>
            <p
              class="person__species"
            >
              <span
                class="person__type"
              >
                species: 
              </span>
              <span
                class="person__text"
              >
                Human
              </span>
            </p>
            <p
              class="person__gender"
            >
              <span
                class="person__type"
              >
                gender: 
              </span>
              <span
                class="person__text"
              >
                Female
              </span>
            </p>
            <p
              class="person__status"
            >
              <span
                class="person__type"
              >
                status: 
              </span>
              <span
                class="person__text"
              >
                Alive
              </span>
            </p>
            <p
              class="person__location"
            >
              <span
                class="person__type"
              >
                location: 
              </span>
              <span
                class="person__text"
              >
                Shiganshina District
              </span>
            </p>
          </div>
        </div>
        <div
          class="person"
        >
          <img
            class="person__img"
            src="https://static.wikia.nocookie.net/shingekinokyojin/images/9/93/Armin_Arlelt_%28Anime%29_character_image.png"
          />
          <div
            class="person__info"
          >
            <h2
              class="person__name"
            >
              Armin Arlelt
            </h2>
            <p
              class="person__species"
            >
              <span
                class="person__type"
              >
                species: 
              </span>
              <span
                class="person__text"
              >
                Human, Intelligent Titan
              </span>
            </p>
            <p
              class="person__gender"
            >
              <span
                class="person__type"
              >
                gender: 
              </span>
              <span
                class="person__text"
              >
                Male
              </span>
            </p>
            <p
              class="person__status"
            >
              <span
                class="person__type"
              >
                status: 
              </span>
              <span
                class="person__text"
              >
                Alive
              </span>
            </p>
            <p
              class="person__location"
            >
              <span
                class="person__type"
              >
                location: 
              </span>
              <span
                class="person__text"
              >
                Shiganshina District
              </span>
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
